stages/org.osbuild.write-device

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/python3
+"""
+Write to device
+
+"""
+
+import fcntl
+import os
+import struct
+import subprocess
+import sys
+
+import osbuild.api
+from osbuild.util import parsing
+
+
+def blksize(device_path):
+    req = 0x80081272  # BLKGETSIZE64, result is bytes as unsigned 64-bit integer (uint64)
+    buf = b' ' * 8
+    fmt = 'L'
+
+    with open(device_path, "rb") as dev:
+        buf = fcntl.ioctl(dev.fileno(), req, buf)
+    n_bytes = struct.unpack(fmt, buf)[0]
+    return n_bytes
+
+
+def main(args, devices, options):
+    src = parsing.parse_location(options.get("from"), args)
+    device = devices["device"]["path"]
+
+    src_size = os.path.getsize(src)
+    print(f"src: '{src}' src_size: {src_size}")
+    device_size = blksize(device)
+    print(f"device: '{device}' device_size: {device_size}")
+    if src_size > device_size:
+        raise ValueError("File to large for device")
```

Comment:
Nitpick: "too large".

Reply:
> (super nitpick) might be nice to include the actual sizes, so `file (xxx mb) to large for device (capacity: yyy)`

you mean `file (xxx mb) **too** large`... :laughing: